connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/streaming/ClientStreamingQuerySuite.scala

```diff
@@ -175,6 +175,37 @@ class ClientStreamingQuerySuite extends QueryTest with SQLHelper with Logging {
     }
   }
 
+  test("throw exception in streaming") {
+    val session = spark
+    import session.implicits._
+
+    val checkForTwo = udf((value: Int) => {
+      if (value == 2) {
+        throw new RuntimeException("Number 2 encountered!")
+      }
+      value
+    })
+
+    val query = spark.readStream
+      .format("rate")
+      .option("rowsPerSecond", "1")
+      .load()
+      .select(checkForTwo($"value").as("checkedValue"))
+      .writeStream
+      .outputMode("append")
+      .format("console")
+      .start()
+
+    val exception = intercept[SparkException] {
+      query.awaitTermination()
+    }
+
+    assert(
+      exception.getCause.getCause.getCause.getMessage
```

Comment:
Remind me again what is the chain here? And why are we not checking this?

Reply:
https://github.com/apache/spark/pull/42377#discussion_r1323060159